### DISCUSSION TITLE
How to access multiple keys together from a json file

### DISCUSSION THREAD

**Discussion by MdAnayat** [nlohmann/json#2252] [Q&A] [answered]
Hello,
First of all thanx for this amazing JSON library. 
I have a JSON file called input.json.
The keys and values inside of the input.json file is as follows:
{"values":"init","roi_x":"10","roi_y":"10","roi_w":"10","roi_h":"10","red_min":"100","red_max":"150"}
Now, for accessing each parameter of the input.json file i have used the below code:

    std::ifstream ifs("./input.json");
    json j = json::parse(ifs);
	for (const auto& element : j["roi_x"])
	{
		x = std::stoi(element.get_ref<const std::string&>());
		std::cout << "The received x_coordinate value: " << x << std::endl;
	}
	for (const auto& element : j["roi_y"])
	{
		y = std::stoi(element.get_ref<const std::string&>());
		std::cout << "The received y_coordinate value: " << y << std::endl;
	}

	for (const auto& element : j["roi_w"])
	{
		w = std::stoi(element.get_ref<const std::string&>());
		std::cout << "The received width of the ROI: " << w << std::endl;
	}

	for (const auto& element : j["roi_h"])
	{
		h = std::stoi(element.get_ref<const std::string&>());
		std::cout << "The received height of the ROI: " << h << std::endl;
	}

	for (const auto& element : j["red_min"])
	{
		r = std::stoi(element.get_ref<const std::string&>());
		std::cout << "The received minimum Red value: " << r << std::endl;
	}

	for (const auto& element : j["red_max"])
	{
		k = std::stoi(element.get_ref<const std::string&>());
		std::cout << "The received maximum red value: " << k << std::endl;
	}

The problem is for accessing each of the parameters of input.json file i need to use for loop.

My question: Is there any way to access all the parameters using the for loop only one time??? I want to refrain myself using for loops so many times.

Any kind of help to solve this issue will be highly appreciated..

Best Regards
Anayat

**Comment by nlohmann** [comment, marked as answer]
You do not need to loop at all. You can use this:

```cpp
x = std::stoi(j["roi_x"].get_ref<const std::string&>());
std::cout << "The received x_coordinate value: " << x << std::endl;

y = std::stoi(j["roi_y"].get_ref<const std::string&>());
std::cout << "The received y_coordinate value: " << y << std::endl;
```

**Reply by MdAnayat** [reply]
It solved my problem. Thanx a lot for your quick response
